src/nf/nf_dense_layer.f90

```diff
@@ -96,6 +98,12 @@ module function get_params(self) result(params)
         !! Parameters of this layer
     end function get_params
 
+    module subroutine get_params_ptr(self, w_ptr, b_ptr)
+      class(dense_layer), intent(in), target :: self
+      real, pointer :: w_ptr(:,:)
```

Comment:
Could this return a 1D array instead of a 2D array? 
This will avoid to have to duplicate the `minimize` procedures

Reply:
I think so, I'll give it a try.